assets/app/scripts/controllers/deployments.js

```diff
@@ -85,6 +120,222 @@ angular.module('openshiftConsole')
       }
     }
 
+    $scope.startLatestDeployment = function(deploymentConfigName) {
+      var deploymentConfig = $scope.deploymentConfigs[deploymentConfigName];
+
+      // increase latest version by one so starts new deployment based on latest
+      var req = {
+        kind: "DeploymentConfig",
+        apiVersion: "v1",
+        metadata: deploymentConfig.metadata,
+        spec: deploymentConfig.spec,
+        status: deploymentConfig.status
+      };
+      if (!req.status.latestVersion) {
+        req.status.latestVersion = 0;
+      }
+      req.status.latestVersion++;
+
+      // update the deployment config
+      DataService.update("deploymentconfigs", deploymentConfigName, req, $scope).then(
+        function() {
+            $scope.alerts = [
+            {
+              type: "success",
+              message: "Deployment #" + req.status.latestVersion + " of " + deploymentConfigName + " has started.",
+            }
+          ];
+        },
+        function(result) {
+          $scope.alerts = [
+            {
+              type: "error",
+              message: "An error occurred while starting the deployment.",
+              details: getErrorDetails(result)
+            }
+          ];
+        }
+      );
+    };
+
+    $scope.retryFailedDeployment = function(deploymentConfigName, deploymentName) {
+      var deployment = $scope.deploymentsByDeploymentConfig[deploymentConfigName][deploymentName];
+      var req = deployment;
+
+      // delete the deployer pod as well as the deployment hooks pods, if any
+      DataService.list("pods", $scope, function(list) {
```

Comment:
Deployment is cancelled by setting the cancel annotation on an RC - it's
out of your hands then.

On Aug 26, 2015, at 10:47 AM, Jessica Forrester <notifications@github.com>
wrote:

In assets/app/scripts/controllers/deployments.js
<https://github.com/openshift/origin/pull/4177#discussion_r37982680>:

> +            {
> +              type: "error",
> +              message: "An error occurred while starting the deployment.",
> +              details: getErrorDetails(result)
> +            }
> +          ];
> +        }
> +      );
> +    };
> +
> +    $scope.retryFailedDeployment = function(deploymentConfigName, deploymentName) {
> +      var deployment = $scope.deploymentsByDeploymentConfig[deploymentConfigName][deploymentName];
> +      var req = deployment;
> +
> +      // delete the deployer pod as well as the deployment hooks pods, if any
> +      DataService.list("pods", $scope, function(list) {

i hate that we have to get the whole list of pods and then find the ones
that match up :-\
@ironcladlou <https://github.com/ironcladlou> is it possible for the RC
generated by the deployment to have an annotation for who its deployer pod
was?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4177/files#r37982680>.